pkg/hashicorp/tfenterprise/organization/handlers.go

```diff
@@ -0,0 +1,89 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package organization
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"github.com/hashicorp/go-tfe"
+	"github.com/mark3labs/mcp-go/mcp"
+	"github.com/mark3labs/mcp-go/server"
+	log "github.com/sirupsen/logrus"
+	"terraform-mcp-server/pkg/hashicorp/tfenterprise/util"
+)
+
+func ListOrganizations(tfeClient *tfe.Client, logger *log.Logger) (tool mcp.Tool, handler server.ToolHandlerFunc) {
+	return mcp.NewTool("listOrganizations",
+			mcp.WithDescription("This tool lists all organizations the authenticated user has access to in Terraform cloud/ enterprise. An optional query can be passed, which helps match organizations by name or email."),
+			mcp.WithTitleAnnotation("List organizations accessible to the authenticated Terraform user"),
+			mcp.WithReadOnlyHintAnnotation(true),
+			mcp.WithOpenWorldHintAnnotation(false),
+			mcp.WithString("query", mcp.Description("Optional: Filter organizations by name or email")),
+		), func(ctx context.Context, request mcp.CallToolRequest) (*mcp.CallToolResult, error) {
+			var query string
+			if queryVal, ok := request.Params.Arguments["query"]; ok {
```

Comment:
nice that the endpoint had a query parameter! We should consider changing the name of the tool to `searchOrganmizations` because of it.